sdk/metric/view/view.go

```diff
@@ -17,22 +17,195 @@
 
 package view // import "go.opentelemetry.io/otel/sdk/metric/view"
 
-// Config contains configuration options for a view.
-type Config struct {
-	// TODO (#2837): implement.
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+)
+
+// View provides users with the flexibility to customize the metrics that are
+// output by the SDK. A View can be used to:
+//
+// * Ignore Instruments.
+// * Change the name of an Instrument.
+// * Change the aggregation of an Instrument.
+// * Customize which attribute(s) are to be reported by the Instrument.
+//
+// An empty config will match all instruments, and do no modifications.
+type View struct {
+	instrumentName *regexp.Regexp
+	hasWildcard    bool
+	scope          instrumentation.Library
+
+	filterKeys  map[attribute.Key]struct{}
+	viewName    string
+	description string
+	// TODO: Aggregation selection
+}
+
+// New returns a new configured view Config.
+func New(opts ...Option) (View, error) {
+	v := View{
+		filterKeys: map[attribute.Key]struct{}{},
+	}
+
+	for _, opt := range opts {
+		v = opt.apply(v)
+	}
+
+	emptyLibrary := instrumentation.Library{}
+	if v.instrumentName == nil &&
+		v.scope == emptyLibrary {
+		return View{}, fmt.Errorf("must provide at least 1 match option")
+	}
```

Comment:
Is this really an error? Shouldn't this match the default, all instruments?

Reply:
Yes, the specification says that it should be an error

> If none the optional criteria is provided, the SDK SHOULD treat it as an error.

Reply:
gotcha, :+1: